### PR TITLE
Add support for word durations.

### PR DESCRIPTION
**Pull request by shaper**

### CODE REVIEW

File: src/lmnt/api.py [hunk 1]
```diff
@@ -95,12 +123,79 @@ async def synthesize(self, text, voice, **kwargs):
...
-    form_data.add_field('format', kwargs.pop('format', 'wav'))
-    form_data.add_field('speed', kwargs.pop('speed', 1.0))
+    form_data.add_field('seed', kwargs.get('seed', 0))
+    form_data.add_field('format', kwargs.get('format', 'wav'))
+    form_data.add_field('speed', kwargs.get('speed', 1.0))
+
+    is_multipart_response = False
+    extras = ''
+
+    has_duration = kwargs.get('durations', False)
```
Comment (zachoverflow) [inline src/lmnt/api.py:133]: I think I would go with "alignment" instead of "durations"?

File: src/lmnt/api.py [hunk 2]
```diff
@@ -95,12 +123,79 @@ async def synthesize(self, text, voice, **kwargs):
...
+      else:
+        await self._handle_response_errors(resp)
+        return await resp.read()
+
+  def _transform_to_word_durations(self, durations, phonemes):
+    if not durations or not phonemes or len(durations) != len(phonemes):
+      raise ValueError("Invalid word transformation input data.")
+
+    result = []
+    temp_phonemes, temp_durations = [], []
```
Comment (zachoverflow) [inline src/lmnt/api.py:159]: nit: acc instead of temp? since we're accumulating

File: src/lmnt/api.py [hunk 3]
```diff
@@ -95,12 +123,79 @@ async def synthesize(self, text, voice, **kwargs):
...
+    total_duration, start = 0, 0
+
+    for index, (dur, phon) in enumerate(zip(durations, phonemes)):
+      temp_phonemes.append(phon)
+      duration_samples = dur * _SAMPLES_PER_FRAME
+      temp_durations.append(duration_samples)
+      total_duration += duration_samples
+
+      # Break words at any whitespace phoneme.
```
Comment (zachoverflow) [inline src/lmnt/api.py:168]: super nit: remove phoneme since whitespace isn't technically a phoneme

File: src/lmnt/api.py [hunk 4]
```diff
@@ -95,12 +123,79 @@ async def synthesize(self, text, voice, **kwargs):
...
+      if phon == " " or ((index < len(durations) - 1) and phonemes[index + 1] == " "):
```
Comment (zachoverflow) [inline src/lmnt/api.py:169]: I think we can eliminate the peek and simplify this if we move before line 163?

e.g.

for all the phones:
  - if current is whitespace:
    - emit current accumulation if anything accumulated
     - emit current current whitespace token
  - else:
    - accumulate

and then after the for, make sure to emit any remaining accumulation (to catch case if last is not whitespace)

File: src/lmnt/api.py [hunk 5]
```diff
@@ -16,12 +16,15 @@
 import aiohttp
 import json
 
+from loguru import logger
```
Comment (zachoverflow) [inline src/lmnt/api.py:19]: I don't see the logger used in this change, so remove?

**Review by zachoverflow — APPROVED** [review]

File: src/lmnt/api.py (hunk 5 above)
Comment (shaper) [reply src/lmnt/api.py:19]: Done. May add it back in future PR once there's logging that feels worth leaving in (I had some, but removed it).

File: src/lmnt/api.py (hunk 1 above)
Comment (shaper) [reply src/lmnt/api.py:133]: Done.

File: src/lmnt/api.py (hunk 2 above)
Comment (shaper) [reply src/lmnt/api.py:159]: Done.

File: src/lmnt/api.py (hunk 3 above)
Comment (shaper) [reply src/lmnt/api.py:168]: Done.

**Comment by sharvil** [comment]
Just a note on conventions: durations are used for a sequence of integers
describing the duration of some underlying unit (e.g. phoneme, grapheme)
whereas alignment is used to describe a matrix (often a binary matrix) with
the time scales on each axis.

I think in this case, durations is probably right.

File: src/lmnt/api.py (hunk 4 above)
Comment (shaper) [reply src/lmnt/api.py]: Done.

**Review by zachoverflow — APPROVED** [review]

**Comment by shaper** [comment]
> Just a note on conventions: durations are used for a sequence of integers describing the duration of some underlying unit (e.g. phoneme, grapheme) whereas alignment is used to describe a matrix (often a binary matrix) with the time scales on each axis. I think in this case, durations is probably right.

LOL, I didn't see this in time -- will put it back, no problem.

**Pull request merged**